lib/rucio/common/hash_algorithms/default.py

```diff
@@ -0,0 +1,110 @@
+import hashlib
+import io
+import mmap
+import zlib
+
+from functools import partial
+
+from . import RucioHashAlgorithm
+
+
+class Adler32(RucioHashAlgorithm):
+    @staticmethod
+    def compute_on_file(file):
+        """
+        An Adler-32 checksum is obtained by calculating two 16-bit checksums A and B
+        and concatenating their bits into a 32-bit integer. A is the sum of all bytes in the
+        stream plus one, and B is the sum of the individual values of A from each step.
+
+        :param file: file name
+        :returns: Hexified string, padded to 8 values.
+        """
```

Comment:
Please include a `:raises ...:` comment.

For reference:https://github.com/rucio/rucio/blob/c9ea1c2dcb2a64f9e2ccac49d021914b863a684e/lib/rucio/client/uploadclient.py#L124-L127

Reply:
Fixed - but that's another bit moved verbatim from rucio/common/utils.py.